src/pages/why-isnt-x-a-hook/index.az.md

```diff
@@ -0,0 +1,238 @@
+
+`React.memo()` bir komponent götütür və bir komponent qaytarır:
+
+```js{4}
+function Button(props) {
+  // ...
+}
+export default React.memo(Button);
+```
+
+**Amma bu niyə sadəcə bir Hook deyil?**
+
+İstəyirsiniz siz bunu `useShouldComponentUpdate()`, `usePure()`, `useSkipRender()`, və ya `useBailout()` adlandırın, təklif bunun kimi bir şeyə bənzəməyə meyl edir:
+
+```js
+function Button({ color }) {
+  // ⚠️ real API deyil
+  useBailout(prevColor => prevColor !== color, color);
+
+  return (
+    <button className={'button-' + color}>  
+      OK
+    </button>
+  )
+}
+```
+
+Bir az daha da variasiyalar vardır (məs. `usePure()` markeri), amma geniş vuruşlarda onların hamsının eyni qüsurları vardır.
+
+### Kompozisiya
+
+Deyək ki, biz `useBailout()`ı iki məxsusi Hookda yoxlayırıq:
+
+```js{4,5,19,20}
+function useFriendStatus(friendID) {
+  const [isOnline, setIsOnline] = useState(null);
+
+  // ⚠️ Real API deyil
+  useBailout(prevIsOnline => prevIsOnline !== isOnline, isOnline);
+
+  useEffect(() => {
+    const handleStatusChange = status => setIsOnline(status.isOnline);
+    ChatAPI.subscribe(friendID, handleStatusChange);
+    return () => ChatAPI.unsubscribe(friendID, handleStatusChange);
+  });
+
+  return isOnline;
+}
+
+function useWindowWidth() {
+  const [width, setWidth] = useState(window.innerWidth);
+  
+  // ⚠️ Real API deyil
+  useBailout(prevWidth => prevWidth !== width, width);
+
+  useEffect(() => {
+    const handleResize = () => setWidth(window.innerWidth);
+    window.addEventListener('resize', handleResize);
+    return () => window.removeEventListener('resize', handleResize);
+  });
+
+  return width;
+}
+```
+
+İndi siz onların hər ikisini eyni komponentdə istifadə etsəniz nə baş verər?
+
+```js{2,3}
+function ChatThread({ friendID, isTyping }) {
+  const width = useWindowWidth();
+  const isOnline = useFriendStatus(friendID);
+  return (
+    <ChatLayout width={width}>
+      <FriendStatus isOnline={isOnline} />
+      {isTyping && 'Typing...'}
+    </ChatLayout>
+  );
+}
+```
+
+O nəyi yenidən render edər?
+
+Əgər hər bir `useBailout()` çağırışının yenilənməni atlama kimi bir gücü varsa, onda `useWindowWidth()`dan olan yenilənmələr `useFriendStatus()` tərəfindən bloklanardı, və ya əksinə. **Bu Hooklar bir-birini qırardı.**
+
+Lakin, əgər `useBailout()`
+
+However, if `useBailout()` was only respected when *all* calls to it inside a single component “agree” to block an update, our `ChatThread` would fail to update on changes to the `isTyping` prop.
+
+Daha da pisi, bütün bu sematikalarla **`ChatThread`ə `useBailout()` çağırmayan hər hansı bir yeni əlavə edilmiş Hook, onu qırardı.**
+
+**Qərar:** 🔴 `useBailout()` kompozisiyanı qırır. Onu Hooka əlavə etmək digər Hooklarda hal yenilənmələrini qırır. Biz APIların [qırılabilən olmamasını](/optimized-for-change/) istəyirik, bu davranış isə bunun tamamilə əksidir.
+
+### Debaqinq
+
+How does a Hook like `useBailout()` affect debugging?
```

Comment:
Can you please extract the unfinished page into a separate PR? Thanks.

Reply:
I just finished it. Sorry for the wait.